src/api/getCircularRoute.ts

```diff
@@ -0,0 +1,35 @@
+import { ref } from "vue";
+import { ConfigEnv } from "./env.config";
+
+const itineraryPointToString = (itineraryPoint: ItineraryPoint) => {
+    let itineraryPointString = "";
+    //loop through itinerary points and add to string if name is empty omit it
+    if (itineraryPoint.name === "") {
+        itineraryPointString = `${itineraryPoint.lon},${itineraryPoint.lat}`;
+    } else {
+        //url encode the name
+        const name = encodeURIComponent(itineraryPoint.name);
+        itineraryPointString = `${itineraryPoint.lon},${itineraryPoint.lat},${name}`;
+    }
+    return itineraryPointString;
+}
+
+export const fetchCircularRouteCSM = async (itineraryPoint: ItineraryPoint, duration: number, distance: number) => {
+    const url = ref("");
+
+    if(duration == -1){
+        url.value = `https://www.cyclestreets.net/api/journey.json?key=${ConfigEnv.apiKey}&itinerarypoints=${itineraryPointToString(itineraryPoint)}&plan=leisure&distance=${distance}`;
+    }
+    if(distance == -1){
+        url.value = `https://www.cyclestreets.net/api/journey.json?key=${ConfigEnv.apiKey}&itinerarypoints=${itineraryPointToString(itineraryPoint)}&plan=leisure&duration=${duration}`;
+    }
```

Comment:
ten url na początku, tam gdze ref("") to dajmy `https://www.cyclestreets.net/api/journey.json?key=${ConfigEnv.apiKey}&itinerarypoints=${itineraryPointToString(itineraryPoint)}&plan=leisure&` a dalej:
```
if(duration == -1){
        url.value += `distance=${distance}`;
    }
    if(distance == -1){
        url.value += `duration=${duration}`;
    }
```